Fix Issue #1339 Add pin icon to left hand navigation menu for chat rows that are pinned

tugbadogan opened the pull request:
cc @jboniface 

### Details
Added pin icon to pinned chats in the left side bar.

### Fixed Issues
Fixes #1339 

### Tests
1. Log into Expensify Cash
2. Pin a chat using the pin icon at the top right of the chat
3. Verify that small pin icon appears on the right hand side of the chat row in the left-hand nav.

### Tested On

- [X] Web
- [X] Mobile Web
- [X] Desktop
- [X] iOS
- [X] Android

### Screenshots
#### Web
<img width="1440" alt="Screenshot 2021-01-28 at 17 05 44" src="https://user-images.githubusercontent.com/3853003/106173230-9e5a5280-618b-11eb-93fe-da35c8bcfec4.png">

#### Mobile Web
<img width="582" alt="Screenshot 2021-01-28 at 17 06 30" src="https://user-images.githubusercontent.com/3853003/106173271-a6b28d80-618b-11eb-8951-d83c3ae17d01.png">

#### Desktop
<img width="1440" alt="Screenshot 2021-01-28 at 17 03 51" src="https://user-images.githubusercontent.com/3853003/106173298-aca86e80-618b-11eb-8422-0f7dbd760cef.png">

#### iOS
![Simulator Screen Shot - iPhone 11 - 2021-01-28 at 17 02 23](https://user-images.githubusercontent.com/3853003/106173333-b7630380-618b-11eb-93bf-1677eac652a6.png)

#### Android
![Screenshot_1611853006](https://user-images.githubusercontent.com/3853003/106173347-bc27b780-618b-11eb-8b32-becb503ce704.png)

tugbadogan committed:
Fix Issue #1339 Add pin icon to left hand navigation menu for chat rows that are pinned

tugbadogan committed:
Fix Issue #1339 Add pin icon to left hand navigation menu for chat rows that are pinned

tugbadogan commented:
I have read the CLA Document and I hereby sign the CLA

shawnborton commented:
This looks great! 

For Android, is there any way to create a little more separation between the ellipsis and the icon? Notice the difference between iOS and Android:

Android | iOS
----- | -----
![image](https://user-images.githubusercontent.com/2319350/106110911-a09cbc80-614b-11eb-97ea-2a75747fa5dc.png) | ![image](https://user-images.githubusercontent.com/2319350/106110930-a7c3ca80-614b-11eb-9422-75fa955eb90d.png)

shawnborton commented:
Possibly we could add `styles.ml1` (which does `marginLeft: 4px`) to the view that wraps the right icons?

Also, could we see what a chat row looks like when you have entered text into the compose field but you haven't sent it yet? Basically like a "draft" message. Thanks!

tugbadogan commented:
Hi Shawn,

Thanks for quick review of my pull request.

> Possibly we could add `styles.ml1` (which does `marginLeft: 4px`) to the view that wraps the right icons?

Sure, let me add and see how it looks. 👍
 
> Also, could we see what a chat row looks like when you have entered text into the compose field but you haven't sent it yet? Basically like a "draft" message. Thanks!

Yes, it doesn't look good. 

<img width="1440" alt="Screenshot 2021-01-28 at 15 59 41" src="https://user-images.githubusercontent.com/3853003/106165102-a1047a00-6182-11eb-9b3e-f52b4ac822ec.png">

I noticed that the pencil icon is using `variables.fontSizeSmall` for width and `variables.iconSizeSmall` for height. I think we can make the icon 20x20 (`variables.iconSizeNormal`) like pin icon. It would look like this:

<img width="1440" alt="Screenshot 2021-01-28 at 16 01 04" src="https://user-images.githubusercontent.com/3853003/106165649-3869cd00-6183-11eb-871d-cd28ea14a172.png">

shawnborton commented:
Agree with all of those thoughts. 

So perhaps we:
- Add `styles.ml1` to the wrapper that contains all of the icons to the right of a chat row
- Inside of this wrapper, we should make the pencil icon use `iconSizeNormal`
- We should also give the pin icon some margin left (maybe `styles.ml2`) to create a gap between the two icons

How does all of that sound?

tugbadogan commented:
> Agree with all of those thoughts.
> 
> So perhaps we:
> 
> * Add `styles.ml1` to the wrapper that contains all of the icons to the right of a chat row
> * Inside of this wrapper, we should make the pencil icon use `iconSizeNormal`
> * We should also give the pin icon some margin left (maybe `styles.ml2`) to create a gap between the two icons
> 
> How does all of that sound?

I have 2 questions
1. Icon module doesn't use style prop. Do you want me to modify the icon module to accept style prop or wrap the `Icon` with a `View` and add style prop there?
https://github.com/Expensify/Expensify.cash/blob/6424cfdbe990b1e2543c9f75181002cad67e8930/src/components/Icon/index.js#L8-L21
2. Do I need to check whether there is a `Pencil` icon to use `styles.ml2` for the `pin` icon? Otherwise there will be extra margin if there is only `pin` icon.

shawnborton commented:
1. I think you can just wrap the icon in a view that has margin
2. Great point, so actually, would it just be easier if the wrapper itself had no margin, but every item inside of the wrapper had a margin-left?

tugbadogan committed:
Fixed pencil icon size and space between icons

tugbadogan committed:
Fixed pencil icon size and space between icons

tugbadogan committed:
Fixed pencil icon size and space between icons

tugbadogan commented:
I added another commit with these changes and updated the screenshots in the first comment.

<img width="1440" alt="Screenshot 2021-01-28 at 17 05 44" src="https://user-images.githubusercontent.com/3853003/106173488-e2e5ee00-618b-11eb-9f15-b27b90226b20.png">

<img width="582" alt="Screenshot 2021-01-28 at 17 06 30" src="https://user-images.githubusercontent.com/3853003/106173504-e6797500-618b-11eb-890a-605c5484073d.png">

<img width="1440" alt="Screenshot 2021-01-28 at 17 03 51" src="https://user-images.githubusercontent.com/3853003/106173512-eaa59280-618b-11eb-8284-c09846cbcfb2.png">

![Simulator Screen Shot - iPhone 11 - 2021-01-28 at 17 02 23](https://user-images.githubusercontent.com/3853003/106173522-eed1b000-618b-11eb-9114-e784178c9a07.png)

![Screenshot_1611853006](https://user-images.githubusercontent.com/3853003/106173536-f2653700-618b-11eb-9fc9-825d5d618ac4.png)

shawnborton commented:
To clarify, is there margin being applied to the left of each icon? I would be curious to see this where each icon (the pin and the pencil) has left margin of 4px, and then another option where each icon has a left margin of 8px.

tugbadogan commented:
Yes, margins are applied to left of each icon. I adding screenshots with different margins for you to compare. I think 8px margin looks better to me👍

Platform | No Margin | 4px left | 8px left
--- | --- | --- | ---
Web | <img width="1440" alt="Screenshot 2021-01-28 at 22 46 53" src="https://user-images.githubusercontent.com/3853003/106208493-6539d680-61bb-11eb-9b54-6a8bd8fdb92c.png"> | <img width="1440" alt="Screenshot 2021-01-28 at 17 05 44" src="https://user-images.githubusercontent.com/3853003/106208530-72ef5c00-61bb-11eb-9789-7a3d959a9f93.png"> | <img width="1440" alt="Screenshot 2021-01-28 at 22 49 37" src="https://user-images.githubusercontent.com/3853003/106208579-826ea500-61bb-11eb-9c92-21d93b7c9c70.png">
Mobile Web | <img width="506" alt="Screenshot 2021-01-28 at 22 47 51" src="https://user-images.githubusercontent.com/3853003/106208992-2eb08b80-61bc-11eb-942d-75666abbc03b.png"> | <img width="582" alt="Screenshot 2021-01-28 at 17 06 30" src="https://user-images.githubusercontent.com/3853003/106209022-3cfea780-61bc-11eb-9119-a8fe64edaf0c.png"> | <img width="499" alt="Screenshot 2021-01-28 at 22 48 54" src="https://user-images.githubusercontent.com/3853003/106209039-45ef7900-61bc-11eb-907f-9f1002d87935.png">
Desktop | <img width="1440" alt="Screenshot 2021-01-28 at 22 45 34" src="https://user-images.githubusercontent.com/3853003/106209127-63bcde00-61bc-11eb-8f5a-b4cf03d1e65f.png"> | <img width="1440" alt="Screenshot 2021-01-28 at 17 03 51" src="https://user-images.githubusercontent.com/3853003/106209141-6ae3ec00-61bc-11eb-9ce4-602a23e582f7.png"> | <img width="1440" alt="Screenshot 2021-01-28 at 22 42 30" src="https://user-images.githubusercontent.com/3853003/106209171-76371780-61bc-11eb-9f9d-845596d168bb.png">
IOS | ![Simulator Screen Shot - iPhone 11 - 2021-01-28 at 22 40 34](https://user-images.githubusercontent.com/3853003/106209287-aa123d00-61bc-11eb-91f9-2cbdd17a8f63.png) | ![Simulator Screen Shot - iPhone 11 - 2021-01-28 at 17 02 23](https://user-images.githubusercontent.com/3853003/106209303-b0081e00-61bc-11eb-8880-681e03929874.png) | ![Simulator Screen Shot - iPhone 11 - 2021-01-28 at 22 40 56](https://user-images.githubusercontent.com/3853003/106209308-b26a7800-61bc-11eb-9519-a791f3f0223c.png)
Android | ![Screenshot_1611873524](https://user-images.githubusercontent.com/3853003/106209431-eba2e800-61bc-11eb-8eee-0ec19238e210.png) | ![Screenshot_1611853006](https://user-images.githubusercontent.com/3853003/106209439-ee054200-61bc-11eb-9ddd-630196349733.png) | ![Screenshot_1611873460](https://user-images.githubusercontent.com/3853003/106209444-f2315f80-61bc-11eb-8bef-1e226690dd4c.png)

tugbadogan committed:
Merge branch 'tugbadogan-add-pin-icon-to-chatlist' of https://github.com/tugbadogan/Expensify.cash into tugbadogan-add-pin-icon-to-chatlist

shawnborton commented:
I agree, 8px looks great so let's go with that option. Thanks so much for exploring!

tugbadogan committed:
Increase left margin to 8px for pencil and pin icons

tugbadogan committed:
Increase left margin to 8px for pencil and pin icons

tugbadogan commented:
Hi @shawnborton ,

I committed the latest changes and changed margin to 8px.

shawnborton commented:
Looks good to me, thanks! Looks like there are some conflicts we need to resolve in `ChatLinkRow.js`, then let's give this a final review and get it merged cc @madmax330

tugbadogan committed:
Merge pull request #1 from Expensify/master

Merge to origin master

tugbadogan committed:
Merge branch 'tugbadogan-add-pin-icon-to-chatlist' of https://github.com/tugbadogan/Expensify.cash into tugbadogan-add-pin-icon-to-chatlist

tugbadogan commented:
Hi, I resolved the conflicts and it's ready to merge now.

shawnborton commented on src/pages/home/sidebar/OptionRow.js:
```diff
@@ -138,9 +139,23 @@ const OptionRow = ({
+                    <View style={styles.ml2}>
+                        <Icon
+                            src={PinCircle}
+                            width={variables.iconSizeNormal}
```
I think you might be able to omit these declarations as `iconSizeNormal` is the default size of each icon. So could you try without `width`, `height`, and `fill` and see if it looks the same?

tugbadogan replied:
Yes, removed these properties and it looks the same 👍

tugbadogan committed:
Remove redundant properties from icons
